lib/base-compiler.js

```diff
@@ -106,6 +107,9 @@ class BaseCompiler {
             result.inputFilename = inputFilename;
             result.stdout = utils.parseOutput(result.stdout, inputFilename);
             result.stderr = utils.parseOutput(result.stderr, inputFilename);
+            if(execOptions.truncateErrorMessages !== null) {
+                result.stderr = result.stderr.slice(0, execOptions.truncateErrorMessages);
```

Comment:
What you could do if you're worried about sending the correct number of lines, is to check here the length of stderr, and send that information separatedly (`result.stderrLength`?) and display that in the UI.

Also, I think we should add a mark to let the user know that the error has been truncated

Reply:
What should the mark be to indicate that the error has been truncated.  

My initial thought was to put a message at the beginning of the output. Something like ```<Error Truncated>```, but should there be something more?

Reply:
A message just like `<Error Truncated>` is what I had in mind, yes. 
I also think that adding it at the end would probably be a good choice.

Reply:
Considering runCompiler is sometimes overridden, isn't it a better idea to put this in the compile function https://github.com/mattgodbolt/compiler-explorer/blob/6235faf899f635a178396ab02649ad557a1c464c/lib/base-compiler.js#L275

Reply:
I agree with @partouf. Compile is a better place for this :)